src/components/Footer.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useContext } from 'react';
+import { TodoContext } from '../contexts/TodoContext';
+import classNames from 'classnames';
+import { FilterType } from '../App';
+
+export interface FooterType {
+  filter: boolean | null;
+  setFilter: React.Dispatch<React.SetStateAction<boolean | null>>;
+  deleteAll: () => void;
+}
+
+export const Footer: React.FC<FooterType> = ({
+  filter,
+  setFilter,
+  deleteAll,
+}) => {
+  const { todos } = useContext(TodoContext);
+
+  const establishFilter = useCallback(
+    (
+      filterName: FilterType,
+      e: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    ) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      switch (filterName) {
+        case FilterType.ALL: {
+          setFilter(null);
+          break;
+        }
+
+        case FilterType.ACTIVE: {
+          setFilter(false);
+          break;
+        }
+
+        case FilterType.COMPLETED: {
+          setFilter(true);
+          break;
+        }
+      }
+    },
+    [setFilter],
+  );
+
+  const todosLength = todos.filter(x => !x.completed && x.id !== 0).length;
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {todosLength} items left
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames('filter__link', {
+            selected: filter === null,
+          })}
+          data-cy="FilterLinkAll"
+          onClick={e => establishFilter(FilterType.ALL, e)}
```

Comment:
If you have FilterType enum why do you need additional value like filter?
Just compare if FilterType equal required field